src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -148,52 +178,66 @@ class Proxied<T> {
   }
 }
 
+async function resolveDeveloperNodeModule(
```

Comment:
I ended up refactoring some of this module resolution stuff when I was originally trying to support `firebase-functions` 2.x and 3.x but I determined that was not feasible.

Still the refactoring remains because I think I made the code more maintainable.